src/wheel/cli/__init__.py

```diff
@@ -31,12 +31,39 @@ def convert_f(args):
     convert(args.files, args.dest_dir, args.verbose)
 
 
+def tags_f(args):
+    from .tags import tags
+
+    names = (
+        tags(
+            wheel,
+            args.python_tag and args.python_tag.split("."),
+            args.abi_tag and args.abi_tag.split("."),
+            args.platform_tag and args.platform_tag.split("."),
+            args.build,
+            args.remove,
+        )
+        for wheel in args.wheel
+    )
+
+    for name in names:
+        print(name)
+
+
 def version_f(args):
     from .. import __version__
 
     print("wheel %s" % __version__)
 
 
+TAGS_HELP = """\
+Make a new wheel with given tags. Any tags unspecified will remain the same.
+Separate multiple tags with a dot. Starting with a dot will append to the
+existing tags. The original file will remain unless --remove is given. The
+output file(s) will be displayed on stdout.
```

Comment:
It won't work out of the box exactly like that. Out of the box, `--python-tag=-py37` will work. We could probably pre-process this to reduce paper cuts, but possibly introduce more issues. Guessing remove tags isn't extremely common - adding would be much more common (and this is a standard argument parsing issue), so IMO it would be fine to go that way.

Reply:
What should happen if you mix and match? `--python-tag=+py37 --python-tag=+py38` is obvious, but what about `--python-tag=py37 --python-tag=py38`?

If you do want to go with this sort of mechanism, IMO, the main thing you need is a clear operation. How about this, actually: Add a flag `--append-tags` which causes the tags to be appended instead of cleared. So the following:

```
Initial wheel:  cmake-3.20.2-py3-none-macosx_10_10_universal2.whl

CLI: wheel tags ... --python-tag py2
Out: cmake-3.20.2-py2-none-macosx_10_10_universal2.whl

CLI: wheel tags ... --python-tag py2 --python-tag py3
Out: cmake-3.20.2-py2.py3-none-macosx_10_10_universal2.whl

CLI: wheel tags ... --append-tags --python-tag py2
Out: cmake-3.20.2-py2.py3-none-macosx_10_10_universal2.whl
```

If a tag is listed one or more times, it is joined and set. If append-tags is given, then tags are added instead of set. Only parts given are touched.

Thoughts?

Reply:
@agronholm Thoughts on the suggestion above?

Reply:
Sorry for not responding earlier.

> but what about --python-tag=py37 --python-tag=py38

Then the Python tag would become `py38`.

> And what if you try to remove a non-existing tag?

That would be a no-op.

> Or one you also give (possible if scripted)?

Then that operation is negated (no-op).

> IMO, the main thing you need is a clear vs. append operation.

Isn't that what I suggested? Setting a tag without modifiers clears all other tags of the same category. Is there something that could not be done with the scheme I proposed?

Reply:
Okay, I've implemented your suggestion. I allow one further feature - if you pass `--python-tag=py37.py38`, that is the equivalent to `--python-tag=py37 --python-tag=+py38`. Having to add +s on all "extra" tags after the first one is IMO is a bit strange for a CLI interface. I'm pretty sure users will be trying `--python-tag=py37 --python-tag=py38` and expecting it to work. Would you like that to be an error, actually? I think, especially for a new interface, having an error would be friendlier than just silently ignoring things the user is explicitly passing.

Reply:
Allowing users to set all tags in one shot is a good idea. What if we take it one step further and only allow the option to be used once? We would then allow the user to either set a specific tag, or pass a comma-separated list of changes like `+py38,-py37`. Attempting to pass tag (re)set options (without + or -) in a comma separated list would be an error.

Reply:
How about allowing only:

```
--python-tag=py37.py38 # set, can take multiple dot separated tags
--python-tag=+py3.py4  # add, can take multiple dot separated tags
```

Tags are traditionally dot separated, the above doesn't add commas, and looks a lot like what the tags end up as.

Do we really need the remove operation? Adding is a common operation and would be nice to have a shortcut, but removing could always be done by setting instead, and setting a lot of tags at once is easy with the syntax above. (Adding a minus means you would want to be able to mix minus and add operations, which could get weird)

Reply:
I would err on the side of caution and keep incremental tag removal as a supported feature.

> Also the leading minus is not supported by argparse when passed without the =.

I don't see that as much of a problem. We should write any examples with the =, and anybody really trying to get tags removed should at least look at the docs to see how it's done in case they run into trouble.

Reply:
"Err on the side of caution" for a new API usually means to keep it minimal, and add features if they are requested, rather than try to add every feature you think someone might need - you can't easily remove a feature later, but you can always add.

How about this: `-a.b.c` will remove a, b, and c, and there isn't a way to remove one tag, and then add two on one operation. Basically, in "explain in the help text" terms, starting with a `+` adds tags, starting with a `-` removes tags. If you want to remove and add, you can either run it twice, or just set the tags to what you need.

Reply:
Also, with add and set you can't get into the situation where there are no tags left. Remove lets you do that, though, so we need some sort of operation or error if you get empty tags, which is not allowed.

Reply:
> How about this: -a.b.c will remove a, b, and c, and there isn't a way to remove one tag, and then add two on one operation. Basically, in "explain in the help text" terms, starting with a + adds tags, starting with a - removes tags. If you want to remove and add, you can either run it twice, or just set the tags to what you need.

I think this would be acceptable. If some great need for doing both these operations in one shot, then we can later add support for that comma separated list.